cmd/argocd-server/commands/argocd_server.go

```diff
@@ -142,7 +144,11 @@ func NewCommand() *cobra.Command {
 
 			dynamicClient := dynamic.NewForConfigOrDie(config)
 
-			controllerClient, err := client.New(config, client.Options{})
```

Comment:
Change is required if user is trying to `generate` appset with non default project (or create with `--dry-run=true` flag)